### PR TITLE
Select fullscreen resolution with max available refresh rate

### PR DESCRIPTION
**Pull request by UselessToucan**
Resolves ppy/osu#2477 (in some cases).

---

### CODE REVIEW

**Comment by peppy** [comment]
Hard crashes for me:

image

```
Unhandled Exception: System.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.Max(IEnumerable`1 source)
   at osu.Framework.Platform.DesktopGameWindow.getMaxRefreshRate(Size size) in C:\Users\Dean\Projects\osu-framework\osu.Framework\Platform\DesktopGameWindow.cs:line 106
```

File: osu.Framework/Platform/DesktopGameWindow.cs [hunk 1]
```diff
@@ -101,6 +103,12 @@ private void changeResolution(Size newSize)
             }
         }
 
+        private float getMaxRefreshRate(Size size)
+        {
+            var displayResolutions = AvailableDisplayResolutions.Where(r => r.Width == size.Width && r.Height == size.Height).ToList();
+            return displayResolutions.Any() ? displayResolutions.Select(r => r.RefreshRate).Max() : DisplayDevice.Default.RefreshRate;
```
Comment (grumd) [inline osu.Framework/Platform/DesktopGameWindow.cs]: maybe use `currentDisplay.RefreshRate` as the default value?

Comment (UselessToucan) [reply osu.Framework/Platform/DesktopGameWindow.cs]: `currentDisplay` is a local variable from the `changeResolution` method with the same `DisplayDevice.Default.RefreshRate` value.

**Comment by FreezyLemon** [comment]
I'd prefer handing more control to the consumer instead of always selecting the highest refresh rate...
For example a `FrameworkSetting.RefreshRate`.

Otherwise, you might want to change the current code a bit:
Why are you going through all resolutions, picking one with the right width and height and then _only_ taking the refresh rate from that one? 
It would make more sense to find (include BitsPerPixel) and then take the right resolution instead of just getting the refresh rate.

File: osu.Framework/Platform/DesktopGameWindow.cs (hunk 1 above)
Comment (grumd) [reply osu.Framework/Platform/DesktopGameWindow.cs]: you could pass the variable from there, but if it's the same value then I guess it doesn't matter

**Pull request merged**